### PR TITLE
Add support for resource type priority classes

### PR DESCRIPTION
**Pull request by kasturinarra**
Add support for resource type priority classes and also adds library which helps to delete a priority class after scenario.

### CODE REVIEW

**Comment by kasturinarra** [comment]
@zhouying7780 can you please help review ? Thanks

File: lib/openshift/priority_class.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require 'openshift/cluster_resource'
```
Comment (pruan-rht) [inline lib/openshift/priority_class.rb]: @kasturinarra FYI, you don't really to add this for deletion if you are calling `admin.cli_exec...`

**Comment by zhouying7780** [comment]
LGTM.

File: features/step_definitions/scheduler.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Given /^the #{QUOTED} scheduler priorityclasses is restored after scenario$/ do |name|
+  ensure_admin_tagged
```
Comment (liangxia) [inline features/step_definitions/scheduler.rb]: Also `ensure_destructive_tagged` so it does not interfere with other scenarios.

**Pull request merged**